muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/actions/main-function/evaluation/evaluationActions.ts

```diff
@@ -0,0 +1,1992 @@
+              ...assignmentsEvaluated,
+              ...assignmentsExcercise,
+            ];
+
+            let materials: Array<MaterialContentNodeType>;
+            let evaluations: Array<MaterialEvaluationType>;
+            [materials, evaluations] = <any>await Promise.all([
+              Promise.all(
+                assignments.map((assignment) => {
+                  return promisify(
+                    mApi().materials.html.read(assignment.materialId),
+                    "callback"
+                  )();
+                })
+              ),
+              Promise.all(
+                assignments.map((assignment) => {
+                  return promisify(
+                    mApi().workspace.workspaces.materials.evaluations.read(
+                      workspaceId,
+                      assignment.id,
+                      {
+                        userEntityId,
+                      }
+                    ),
+                    "callback"
+                  )().then((evaluations: Array<MaterialEvaluationType>) => {
+                    return evaluations[0];
+                  });
+                })
+              ),
+            ]);
+
+            return materials.map((material, index) => {
+              return <MaterialContentNodeType>Object.assign(material, {
+                evaluation: evaluations[index],
+                assignment: assignments[index],
+              });
+            });
+          })(),
+        ]);
+
+        dispatch({
+          type: "UPDATE_EVALUATION_RECORDS_CURRENT_STUDENT",
+          payload: {
+            materials,
+          },
+        });
+
+        dispatch({
+          type: "UPDATE_CURRENT_SELECTED_EVALUATION_DATA_STATE",
+          payload: <EvaluationStateType>"READY",
+        });
+      } catch (err) {
+        if (!(err instanceof MApiError)) {
+          throw err;
+        }
+        dispatch(
+          notificationActions.displayNotification(
+            state.i18n.text.get(
+              "plugin.evaluation.notifications.assignmentdata.error",
+              err.message
+            ),
+            "error"
+          )
+        );
+
+        dispatch({
+          type: "UPDATE_CURRENT_SELECTED_EVALUATION_DATA_STATE",
+          payload: <EvaluationStateType>"ERROR",
+        });
+      }
+    };
+  };
+
+/**
+ * setSelectedWorkspaceId
+ * @param data
+ */
+let setSelectedWorkspaceId: SetEvaluationSelectedWorkspace =
+  function setSelectedWorkspaceId(data) {
+    return async (
+      dispatch: (arg: AnyActionType) => any,
+      getState: () => StateType
+    ) => {
+      dispatch({
+        type: "SET_EVALUATION_SELECTED_WORKSPACE",
+        payload: data.workspaceId,
+      });
+
+      dispatch(loadEvaluationSortFunctionFromServer());
+
+      dispatch(loadEvaluationAssessmentRequestsFromServer());
+    };
+  };
+
+/**
+ * setEvaluationFilters
+ * @param data
+ */
+let setEvaluationFilters: SetEvaluationFilters = function setEvaluationFilters(
```

Comment:
Can we not use let anymore if we are not modifying the variables, in the older code you see a lot of let, because a lot was renamed from var just by replacing, and also const was not working for some time for some reason (either misconfiguration or something); we should use const now with newer code, and aim removing all the older lets when we have the chance.

Reply:
Fixed. I though there was reason other that var conversion why let where used with some specific components or with redux implementation